pyinter/interval_set.py

```diff
@@ -84,6 +89,13 @@ def add(self, other):
         This has no effect if the Interval is already represented.
         :param other: an Interval to add to this IntervalSet.
         """
+        # If the other is a collection of intervals add them one by one.
+        try:
+            for inter in other:
+                self.add(inter)
+            return
+        except TypeError:
```

Comment:
I don't like using exceptions for control flow unless absolutely necessary but I'm not sure what the best alternative is here. Maybe instead of calling `add` in `__sub__` we can do an instance check and call `add` or `union` depending on the result.
What do you think?

I've been trying as far as possible to stick to the inbuilt [`set` api](https://docs.python.org/2/library/stdtypes.html#set-types-set-frozenset).